dashboard/app/controllers/ai_tutor_interactions_controller.rb

```diff
@@ -81,4 +81,21 @@ def index
     end
     render json: student_chats
   end
+
+  # GET /ai_tutor_interactions/get_for_student
+  def get_for_student
+    params.require([:userId])
+    student = User.find(params[:userId])
+    return render(status: :not_found, json: {error: 'Student not found'}) unless student
+    teacher_can_view_chats = current_user.students.include?(student)
+    student_can_view_own_chats = student.id == current_user.id
+    return render(status: :forbidden, json: {error: 'This user does not have access to these chats'}) unless teacher_can_view_chats || student_can_view_own_chats
```

Comment:
can/should this logic go in `ability.rb`? We do similar logic around code review comments there [link](https://github.com/code-dot-org/code-dot-org/blob/33df8e52769132afdefd02af06e5bda3dae775ec/dashboard/app/models/ability.rb#L144)